common/djangoapps/request_cache/middleware.py

```diff
@@ -58,3 +58,54 @@ def process_exception(self, request, exception):  # pylint: disable=unused-argum
         """
         self.clear_request_cache()
         return None
+
+
+def request_cached(f):
+    """
+    A decorator for wrapping a function and automatically handles caching its return value, as well as returning
+    that cached value for subsequent calls to the same function, with the same parameters, within a given request.
+
+    Notes:
+        - we convert arguments and keyword arguments to their string form to build the cache key, so if you have
+          args/kwargs that can't be converted to strings, you're gonna have a bad time (don't do it)
+        - cache key cardinality depends on the args/kwargs, so if you're caching a function that takes five arguments,
+          you might have deceptively low cache efficiency.  prefer function with fewer arguments.
+        - we use the default request cache, not a named request cache (this shouldn't matter, but just mentioning it)
+        - benchmark, benchmark, benchmark! if you never measure, how will you know you've improved? or regressed?
+
+    Arguments:
+        f (func): the function to wrap
+
+    Returns:
+        func: a wrapper function which will call the wrapped function, passing in the same args/kwargs,
+              cache the value it returns, and return that cached value for subsequent calls with the
+              same args/kwargs within a single request
+    """
+    def wrapper(*args, **kwargs):
+        """
+        Wrapper function to decorate with.
+        """
+
+        # Build our cache key based on the module the function belongs to, the functions name, and a stringified
+        # list of arguments and a query string-style stringified list of keyword arguments.
+        converted_args = map(str, args)
+        converted_kwargs = map(str, reduce(list.__add__, map(list, kwargs.iteritems()), []))
+        cache_keys = [f.__module__, f.func_name] + converted_args + converted_kwargs
+        cache_key = '.'.join(cache_keys)
+
+        # Check to see if we have a result in cache.  If not, invoke our wrapped
+        # function.  If the result of that function call isn't None, then cache it.
```

Comment:
@tobz: Why isn't `None` cached?  Are you assuming that a result of `None` means that it might just not be ready to give a proper result and a subsequent call will return a different valid result?  Is this a scenario you ran into?

I could imagine scenarios where it takes a lot of processing to arrive at the `None` result, but I might be missing something regarding Python conventions or otherwise.